### PR TITLE
Store exception generated when we tried to update the queue last time

### PR DESCRIPTION
**Pull request by nvartolomei**
The use case is to alert when queue contains broken entries. Especially
important when ClickHouse breaks backwards compatibility between
versions and log entries written by newer versions aren't parseable by
old versions.

```
Code: 27, e.displayText() = DB::Exception: Cannot parse input: expected 'quorum: ' before: 'merge_type: 2\n'
```

I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add new column `last_queue_update_exception` to `system.replicas` table.

### CODE REVIEW

File: src/Storages/StorageReplicatedMergeTree.cpp [hunk 1]
```diff
@@ -3087,23 +3087,26 @@ void StorageReplicatedMergeTree::queueUpdatingTask()
         last_queue_update_finish_time.store(time(nullptr));
         queue_update_in_progress = false;
     }
-    catch (const Coordination::Exception & e)
+    catch (...)
     {
+        last_queue_update_exception.set(std::make_unique<String>(getCurrentExceptionMessage(false)));
```
Comment (alesapin) [inline src/Storages/StorageReplicatedMergeTree.cpp:3092]: Just leave old logic and copy this line to the `catch (...)`.

File: src/Storages/StorageReplicatedMergeTree.cpp [hunk 2]
```diff
@@ -3087,23 +3087,26 @@ void StorageReplicatedMergeTree::queueUpdatingTask()
...
         tryLogCurrentException(log, __PRETTY_FUNCTION__);
 
-        if (e.code == Coordination::Error::ZSESSIONEXPIRED)
+        try
         {
-            restarting_thread.wakeup();
-            return;
+            throw;
```
Comment (alesapin) [inline src/Storages/StorageReplicatedMergeTree.cpp:3097]: ??? Remove.

File: src/Storages/StorageReplicatedMergeTree.h [hunk 3]
```diff
@@ -329,6 +330,7 @@ class StorageReplicatedMergeTree final : public shared_ptr_helper<StorageReplica
     ReplicatedMergeTreeQueue queue;
     std::atomic<time_t> last_queue_update_start_time{0};
     std::atomic<time_t> last_queue_update_finish_time{0};
+    MultiVersion<String> last_queue_update_exception;
```
Comment (alesapin) [inline src/Storages/StorageReplicatedMergeTree.h:333]: `MultiVersion` is not intended to be something like `std::atomic<AnyClass>`. Why not just string with mutex (or even reuse some mutex...) + two methods?

**Review by alesapin — APPROVED** [review]
Ok, let's rerun CI

**Review by alesapin — APPROVED** [review]

**Comment by alesapin** [comment]
@Mergifyio update

**Comment by alesapin** [comment]
@Mergifyio update

**Comment by alesapin** [comment]
Performance failures not related to changes.

**Comment by sevirov** [comment]
Internal documentation ticket: DOCSUP-13565

**Pull request merged**